apps/dashboard/src/app/(dashboard)/event/[id]/edit-card.tsx

```diff
@@ -1,17 +1,20 @@
+import { EventEditSchema } from "@dotkomonline/types"
 import { FC } from "react"
-import { useEventWriteForm } from "../write-form"
-import { EventWriteSchema } from "@dotkomonline/types"
-import { useEventDetailsContext } from "./provider"
 import { useEditEventMutation } from "../../../../modules/event/mutations/use-edit-event-mutation"
+import { useEventEditForm } from "../edit-form"
+import { useEventDetailsContext } from "./provider"
 
 export const EventEditCard: FC = () => {
   const { event } = useEventDetailsContext()
   const edit = useEditEventMutation()
-  const FormComponent = useEventWriteForm({
+  const FormComponent = useEventEditForm({
     label: "Oppdater arrangement",
     onSubmit: (data) => {
-      const result = EventWriteSchema.required({ id: true }).parse(data)
-      edit.mutate(result)
+      const result = EventEditSchema.parse(data)
```

Comment:
I agree with Mats that there should be a check in EventService to ensure the incoming data is valid and throw and error if not. However, you could also add an check in the submit handler which sets the field to dirty on error.